packages/monaco/src/browser/monaco-editor.ts

```diff
@@ -181,14 +183,37 @@ export class MonacoEditor implements TextEditor, IEditorReference {
         return this.onSelectionChangedEmitter.event;
     }
 
-    revealPosition(raw: Position): void {
+    revealPosition(raw: Position, options: RevealPositionOptions = { vertical: 'center', horizontal: 'reveal' }): void {
         const position = this.p2m.asPosition(raw);
-        this.editor.revealPositionInCenter(position);
+        switch (options.vertical) {
+            case 'auto':
+                this.editor.revealPosition(position, false, options.horizontal === 'reveal');
```

Comment:
looks like `options.horizontal` can be boolean, other than that all cases seems to be covered now

Reply:
✔️